code/test/security.js

```diff
@@ -0,0 +1,214 @@
+  // Too early: Can't increase the limit twice within the same interval (ss.lastLimitAdjustmentTime + ss.spendingInterval > block.timestamp && newLimit > ss.spendingLimit)
+  // Too much : Can't increase the limit by more than double existing limit + 1 native Token (newLimit > (ss.spendingLimit) * 2 + (1 ether))
+  it('CO-BASIC-24 CHANGE_SPENDING_LIMIT: must be able to transfer native currency', async () => {
+    // create wallets and token contracts used througout the tests
+    let { walletInfo: alice, state } = await TestUtil.makeWallet({ salt: 'CO-BASIC-24-1', deployer: accounts[0], effectiveTime: getEffectiveTime(), duration: DURATION })
+
+    // Begin Tests
+    let testTime = Date.now()
+
+    testTime = await TestUtil.bumpTestTime(testTime, 60)
+    // alice changes the spending limit
+    let { tx, currentState } = await executeSecurityTransaction(
+      {
+        ...ONEConstants.NullOperationParams, // Default all fields to Null values than override
+        walletInfo: alice,
+        operationType: ONEConstants.OperationType.CHANGE_SPENDING_LIMIT,
+        amount: THREE_ETH,
+        testTime
+      }
+    )
+
+    // Validate succesful event emitted
+    TestUtil.validateEvent({ tx, expectedEvent: 'SpendingLimitChanged' })
+    // TestUtil.validateEvent({ tx, expectedEvent: 'HighestSpendingLimitChanged' })
+
+    // Alice Items that have changed - nonce, lastOperationTime, commits, spendingState
+    state = await TestUtil.validateOpsStateMutation({ wallet: alice.wallet, state })
+    // Spending State
+    const currentSpendingState = await TestUtil.getSpendingStateParsed(alice.wallet) // retrieve this  as it will have lastLimitAdjustmentTime
+    const expectedSpendingState = state.spendingState
+    expectedSpendingState.spendingLimit = THREE_ETH
+    expectedSpendingState.highestSpendingLimit = THREE_ETH
+    expectedSpendingState.lastLimitAdjustmentTime = currentSpendingState.lastLimitAdjustmentTime
+    state.spendingState = await TestUtil.validateSpendingStateMutation({ expectedSpendingState, wallet: alice.wallet })
+    await TestUtil.assertStateEqual(state, currentState)
+  })
+
+  // ====== JUMP_SPENDING_LIMIT ======
+  // Test transferring native currency
+  // Expected result alice can transfer funds to bob
+  // Jump Logic:
+  // Too Much : Can't increase the limit greater than the highest spending limit (newLimit > ss.highestSpendingLimit)
+  // Authentication: from function authenticate in reveal.sol
+  // if innerCores are empty, this operation (in this case) is doomed to fail. This is intended. Client should warn the user not to lower the limit too much if the wallet has no innerCores (use Extend to set first innerCores). Client should also advise the user the use Recovery feature to get their assets out, if they are stuck with very low limit and do not want to wait to double them each spendInterval.
+  it('CO-BASIC-25 JUMP_SPENDING_LIMIT: must be able to transfer native currency', async () => {
```

Comment:
JUMP_SPENDING_LIMIT requires more authentication. Can't use the standard execution logic. See spendLimit.js

Reply:
Done 🙏